rust_src/src/multibyte.rs

```diff
@@ -88,6 +93,10 @@ pub const MAX_MULTIBYTE_LENGTH: usize = 5;
 
 const BYTE8_OFFSET: u32 = 0x3F_FF00;
 
+static mut string_index_cache: LispObject = Qnil;
```

Comment:
This is marked staticpro in the C source. Is that still relevant here?

Reply:
I actually don't know, but I don't imagine it is because as far as I can tell you only use the cached LispObject if you make a second call on the same string, and if you do that it has to be alive anyway, but maybe I'm misunderstanding what staticpro does.

Reply:
My understanding it is means "protected static" and prevents the items from being garbage collected.

Reply:
In that case I think my reasoning for not needing it applies. In case it doesn't, where should I invoke `staticpro`? I've grepped around the codebase and can't find an obvious place to put it.

Reply:
What you are missing is a Rust version of `syms_of_fns`. Looks at the Rust syms_of_cmds. For a while we had syms_of_cmds_rust (or was it rust_syms_of_cmds...).

Reply:
I've added a `rust_syms_of_multibyte` with the staticpro invocation and am calling it from `syms_of_fns`. Since the original function was in fns but the new one in multibyte this seemed like the best available option. Let me know if you want me to change names/locations around.